bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -1197,7 +1177,13 @@ journalFormatVersionToWrite, getBufferedChannelBuilder(),
 
                 toFlush.add(qe);
                 numEntriesToFlush++;
-                qe = null;
+
+                if (localQueueEntriesIdx < localQueueEntriesLen) {
+                    qe = localQueueEntries[localQueueEntriesIdx++];
+                } else {
+                    Arrays.fill(localQueueEntries, 0, localQueueEntriesLen, null);
```

Comment:
Same question with above. At least we can do it in the same loop when you take the queue from the array.